src/puppetlabs/puppetdb/query/events.clj

```diff
@@ -50,49 +50,74 @@
 (defn with-latest-events
   "CTE to wrap unioned queries when distinct_resources is used"
   [query]
-  (format
-    "WITH latest_events AS
-           (SELECT certnames.certname,
-                   resource_events.resource_type,
-                   resource_events.resource_title,
-                   resource_events.property,
-                   MAX(resource_events.timestamp) AS timestamp
-                   FROM resource_events
-                   JOIN certnames ON resource_events.certname_id = certnames.id
-                   WHERE resource_events.timestamp >= ?
-                   AND resource_events.timestamp <= ?
-                   GROUP BY certname, resource_type, resource_title, property)
-           %s" query))
+  (let [collate (if (sutils/postgres?) "COLLATE \"C\"" "")]
+    (format
+      "WITH latest_events AS
+       (SELECT certname,
+       configuration_version,
+       start_time as run_start_time,
+       end_time as run_end_time,
+       receive_time as report_receive_time,
+       hash,
+       status,
+       distinct_events.timestamp AS timestamp,
+       distinct_events.resource_type AS resource_type,
+       distinct_events.resource_title AS resource_title,
+       distinct_events.property as property,
+       new_value,
+       old_value,
+       message,
+       file,
+       line,
+       containment_path,
+       containing_class,
+       name
+       FROM
+       (SELECT certname_id,
+       resource_type %s AS resource_type,
+       resource_title %s AS resource_title,
+       property %s AS property,
+       MAX(resource_events.timestamp) AS timestamp
+       FROM resource_events
+       WHERE resource_events.timestamp >= ?
+       AND resource_events.timestamp <= ?
+       GROUP BY certname_id,
+       resource_type %s,
```

Comment:
I'm not understanding why we would want to collate by ASCII rules here. These come from Puppet as UTF-8 and we store them as UTF-8, what happens when the user has a special character here?